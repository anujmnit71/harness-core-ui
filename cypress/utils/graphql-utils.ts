export const hasOperationName: (req: any, operationName: string) => boolean = (req, operationName) => {
  const { body } = req
  // eslint-disable-next-line no-prototype-builtins
  return body.hasOwnProperty('operationName') && body.operationName === operationName
}

export const aliasQuery: (req: any, operationName: string) => void = (req, operationName) => {
  if (hasOperationName(req, operationName)) {
    req.alias = `gql${operationName}Query`
  }
}
