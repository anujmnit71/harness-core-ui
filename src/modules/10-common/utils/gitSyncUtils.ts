/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { GitSyncConfig } from 'services/cd-ng'
import { folderPathName, yamlFileExtension } from './StringUtils'

export const getRepoDetailsByIndentifier = (identifier: string | undefined, repos: GitSyncConfig[]) => {
  return repos.find((repo: GitSyncConfig) => repo.identifier === identifier)
}

export const validateFilePath = (filePath: string): boolean => {
  if (!filePath.endsWith(yamlFileExtension)) {
    return false
  } else {
    const fullPath = filePath.slice(0, filePath.length - yamlFileExtension.length)
    const subPaths = fullPath.split('/')
    return subPaths.every((folderName: string) => {
      return folderName.length && folderName.match(folderPathName)?.length
    })
  }
}
