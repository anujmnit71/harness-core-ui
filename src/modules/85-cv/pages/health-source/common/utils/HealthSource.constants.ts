/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export const HealthSourceFieldNames = {
  MANUAL_QUERY: 'isManualQuery',
  IDENTIFIER: 'identifier',
  METRIC_NAME: 'metricName',
  METRIC_DATA: 'metricData',
  GROUP_NAME: 'groupName',
  CONTINUOUS_VERIFICATION: 'continuousVerification',
  HEALTH_SCORE: 'healthScore',
  SLI: 'sli',
  RISK_CATEGORY: 'riskCategory',
  HIGHER_BASELINE_DEVIATION: 'higherBaselineDeviation',
  LOWER_BASELINE_DEVIATION: 'lowerBaselineDeviation'
}
