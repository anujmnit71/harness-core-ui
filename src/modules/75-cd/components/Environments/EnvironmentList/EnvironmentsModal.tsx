/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import {
  Button,
  ButtonVariation,
  Formik,
  FormikForm,
  Label,
  Layout,
  VisualYamlSelectedView as SelectedView,
  VisualYamlToggle,
  Container,
  ThumbnailSelect
} from '@wings-software/uicore'
import * as Yup from 'yup'
import { omit } from 'lodash-es'
import { useParams } from 'react-router-dom'
import { Classes } from '@blueprintjs/core'
import type { FormikProps } from 'formik'
import { parse } from 'yaml'
import cx from 'classnames'
import { EnvironmentRequestDTO, EnvironmentResponseDTO, useUpsertEnvironmentV2 } from 'services/cd-ng'
import { IdentifierSchema, NameSchema } from '@common/utils/Validation'
import { getScopeFromDTO } from '@common/components/EntityReference/EntityReference'
import { useGetSchemaYaml } from 'services/pipeline-ng'
import { NameIdDescriptionTags, PageSpinner } from '@common/components'
import { useStrings } from 'framework/strings'
import YAMLBuilder from '@common/components/YAMLBuilder/YamlBuilder'
import type { YamlBuilderHandlerBinding, YamlBuilderProps } from '@common/interfaces/YAMLBuilderProps'
import { useToaster } from '@common/exports'
import css from './EnvironmentsList.module.scss'

interface NewEditEnvironmentModalProps {
  isEdit: boolean
  isEnvironment: boolean
  data: EnvironmentResponseDTO
  envIdentifier?: string
  onCreateOrUpdate(data: EnvironmentRequestDTO): void
  closeModal?: () => void
}

const yamlBuilderReadOnlyModeProps: YamlBuilderProps = {
  fileName: `environment.yaml`,
  entityType: 'Environment',
  width: '100%',
  height: 220,
  showSnippetSection: false,
  yamlSanityConfig: {
    removeEmptyString: false,
    removeEmptyObject: false,
    removeEmptyArray: false
  }
}

export const NewEditEnvironmentModalYaml: React.FC<NewEditEnvironmentModalProps> = ({
  isEdit,
  data,
  isEnvironment,
  onCreateOrUpdate,
  closeModal
}): JSX.Element => {
  const { getString } = useStrings()
  const inputRef = React.useRef<HTMLInputElement | null>(null)
  const { accountId, projectIdentifier, orgIdentifier } = useParams<{
    orgIdentifier: string
    projectIdentifier: string
    accountId: string
  }>()
  const [yamlHandler, setYamlHandler] = React.useState<YamlBuilderHandlerBinding | undefined>()
  const [selectedView, setSelectedView] = React.useState<SelectedView>(SelectedView.VISUAL)
  const { loading: updateLoading, mutate: updateEnvironment } = useUpsertEnvironmentV2({
    queryParams: {
      accountIdentifier: accountId
    }
  })
  const { showSuccess, showError, clear } = useToaster()
  const onSubmit = React.useCallback(
    async (values: EnvironmentRequestDTO) => {
      try {
        if (!values.name) {
          showError(getString('fieldRequired', { field: 'Environment' }))
        } else if (!values.identifier) {
          showError(getString('common.validation.fieldIsRequired', { name: 'Identifier' }))
        } else if (!values.type) {
          showError(getString('fieldRequired', { field: 'Type' }))
        } else if (isEdit && id !== values.identifier) {
          showError(getString('cd.editIdError', { id: id }))
        } else if (isEdit && !isEnvironment) {
          const response = await updateEnvironment({
            ...omit(values, 'accountId', 'deleted'),
            orgIdentifier,
            projectIdentifier
          })
          if (response.status === 'SUCCESS') {
            clear()
            showSuccess(getString('cd.environmentUpdated'))
            onCreateOrUpdate(values)
          }
        } else {
          const response = await updateEnvironment({ ...values, orgIdentifier, projectIdentifier })
          if (response.status === 'SUCCESS') {
            clear()
            showSuccess(getString('cd.environmentCreated'))
            onCreateOrUpdate(values)
          }
        }
      } catch (e) {
        showError(e?.data?.message || e?.message || getString('commonError'))
      }
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [onCreateOrUpdate, orgIdentifier, projectIdentifier, isEdit, isEnvironment]
  )
  React.useEffect(() => {
    inputRef.current?.focus()
  }, [])
  const typeList: { label: string; value: string }[] = [
    {
      label: getString('production'),
      value: 'Production'
    },
    {
      label: getString('nonProduction'),
      value: 'PreProduction'
    }
  ]
  const formikRef = React.useRef<FormikProps<EnvironmentResponseDTO>>()
  const id = data.identifier
  const { data: environmentSchema } = useGetSchemaYaml({
    queryParams: {
      entityType: 'Environment',
      projectIdentifier,
      orgIdentifier,
      accountIdentifier: accountId,
      scope: getScopeFromDTO({ accountIdentifier: accountId, orgIdentifier, projectIdentifier })
    }
  })
  const handleModeSwitch = React.useCallback(
    (view: SelectedView) => {
      if (view === SelectedView.VISUAL) {
        const yaml = yamlHandler?.getLatestYaml() || /* istanbul ignore next */ ''
        const envSetYamlVisual = parse(yaml).environmentInputSet as EnvironmentResponseDTO
        if (envSetYamlVisual) {
          data.name = envSetYamlVisual.name || ''
          data.identifier = envSetYamlVisual.identifier || ''
          data.description = envSetYamlVisual.description || ''
          data.tags = envSetYamlVisual.tags || {}
          data.type = envSetYamlVisual.type

          formikRef.current?.setValues({
            ...omit(data)
          })
        }
      }
      setSelectedView(view)
    },
    [yamlHandler?.getLatestYaml, data]
  )
  if (updateLoading) {
    return <PageSpinner />
  }
  return (
    <>
      <Container className={css.yamlToggleEnv}>
        <Layout.Horizontal flex={{ justifyContent: 'flex-start' }} padding-top="8px">
          <VisualYamlToggle
            selectedView={selectedView}
            onChange={nextMode => {
              handleModeSwitch(nextMode)
            }}
          />
        </Layout.Horizontal>
      </Container>
      <Layout.Vertical>
        <Formik<EnvironmentRequestDTO>
          initialValues={data as EnvironmentRequestDTO}
          enableReinitialize={false}
          formName="deployEnv"
          onSubmit={values => {
            onSubmit(values)
          }}
          validationSchema={Yup.object().shape({
            name: NameSchema({ requiredErrorMsg: getString?.('fieldRequired', { field: 'Environment' }) }),
            type: Yup.string().required(getString?.('fieldRequired', { field: 'Type' })),
            identifier: IdentifierSchema()
          })}
        >
          {formikProps => {
            // @ts-ignore
            formikRef.current = formikProps
            return (
              <>
                {selectedView === SelectedView.VISUAL ? (
                  <FormikForm>
                    <NameIdDescriptionTags
                      formikProps={formikProps}
                      identifierProps={{
                        inputLabel: getString('name'),
                        inputGroupProps: {
                          inputGroup: {
                            inputRef: ref => (inputRef.current = ref)
                          }
                        },
                        isIdentifierEditable: !isEdit
                      }}
                    />
                    <Layout.Vertical spacing={'small'} style={{ marginBottom: 'var(--spacing-medium)' }}>
                      <Label className={cx(Classes.LABEL, css.label)}>{getString('envType')}</Label>
                      <ThumbnailSelect className={css.thumbnailSelect} name={'type'} items={typeList} />
                    </Layout.Vertical>
                    <Layout.Horizontal spacing="small" padding={{ top: 'xlarge' }}>
                      <Button
                        variation={ButtonVariation.PRIMARY}
                        type={'submit'}
                        text={getString('save')}
                        data-id="environment-save"
                      />
                      &nbsp;
                      <Button variation={ButtonVariation.TERTIARY} text={getString('cancel')} onClick={closeModal} />
                    </Layout.Horizontal>
                  </FormikForm>
                ) : (
                  <Container>
                    <YAMLBuilder
                      {...yamlBuilderReadOnlyModeProps}
                      existingJSON={{
                        environmentInputSet: {
                          ...omit(formikProps?.values),
                          description: formikProps.values.description || '',
                          tags: formikProps.values.tags || {},
                          type: formikProps.values.type || ''
                        }
                      }}
                      schema={environmentSchema?.data}
                      bind={setYamlHandler}
                      showSnippetSection={false}
                    />

                    <Layout.Horizontal padding={{ top: 'large' }}>
                      <Button
                        variation={ButtonVariation.PRIMARY}
                        type="submit"
                        text={getString('save')}
                        onClick={() => {
                          const latestYaml = yamlHandler?.getLatestYaml() || /* istanbul ignore next */ ''
                          onSubmit(parse(latestYaml)?.environmentInputSet)
                        }}
                      />
                      &nbsp; &nbsp;
                      <Button variation={ButtonVariation.TERTIARY} onClick={closeModal} text={getString('cancel')} />
                    </Layout.Horizontal>
                  </Container>
                )}
              </>
            )
          }}
        </Formik>
      </Layout.Vertical>
    </>
  )
}
