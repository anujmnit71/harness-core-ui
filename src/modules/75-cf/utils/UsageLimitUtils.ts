/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { ReactNode } from 'react'
import { BannerType } from '@common/layouts/Constants'
import type { CheckFeatureReturn } from 'framework/featureStore/featureStoreUtil'
import { formatToCompactNumber } from '@cf/utils/CFUtils'
import type { UseStringsReturn } from 'framework/strings'

export const getBannerText = (
  getString: UseStringsReturn['getString'],
  monthlyActiveUsers: CheckFeatureReturn,
  additionalLicenseProps: Record<string, any>
): ReactNode => {
  const { isEnterpriseEdition, isFreeEdition, isTeamEdition } = additionalLicenseProps

  const clientMauUsageCount = Number(monthlyActiveUsers?.featureDetail?.count)
  const clientMauPlanLimit = Number(monthlyActiveUsers?.featureDetail?.limit)

  const clientMauUsagePercentage = Math.trunc((clientMauUsageCount / clientMauPlanLimit) * 100)
  const clientMauPlanLimitFormatted = formatToCompactNumber(clientMauPlanLimit)

  const showInfoBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 90 && clientMauUsagePercentage < 100
  const showWarningBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 100

  let message!: string
  let bannerType!: BannerType

  if (isFreeEdition) {
    if (!showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.freePlan.approachingLimit', {
            clientMauUsagePercentage,
            clientMauPlanLimitFormatted
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: getString('cf.planEnforcement.freePlan.upgradeRequired', {
          clientMauPlanLimitFormatted
        }),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  if (isTeamEdition || isEnterpriseEdition) {
    if (showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.teamEnterprisePlan.approachingLimit', {
            clientMauUsagePercentage
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: () => getString('cf.planEnforcement.teamEnterprisePlan.upgradeRequired'),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  return {
    message,
    bannerType
  }
}
