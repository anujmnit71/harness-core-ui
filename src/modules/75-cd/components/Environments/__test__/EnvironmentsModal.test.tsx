/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, getByText } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { environmentModal } from '@cd/mock'
import { NewEditEnvironmentModalYaml } from '../EnvironmentList/EnvironmentsModal'

jest.mock('services/pipeline-ng', () => {
  return {
    useGetSchemaYaml: jest.fn(() => ({ data: null }))
  }
})

describe('Environments', () => {
  test('should render Environments', () => {
    const { container } = render(
      <TestWrapper
        path="account/:accountId/cd/orgs/:orgIdentifier/projects/:projectIdentifier/environment"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <NewEditEnvironmentModalYaml {...environmentModal} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
    fireEvent.click(getByText(document.body, 'save'))
    fireEvent.click(getByText(document.body, 'YAML'))
    fireEvent.click(getByText(document.body, 'save'))
    expect(container).toMatchSnapshot()
  })
})
