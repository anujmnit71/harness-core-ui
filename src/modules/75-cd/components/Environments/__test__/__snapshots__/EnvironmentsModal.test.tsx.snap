// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environments should render Environments 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main yamlToggleEnv"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-start"
      padding-top="8px"
    >
      <div
        class="PillToggle--optionBtns"
      >
        <div
          class="PillToggle--item PillToggle--selected"
          data-name="toggle-option-one"
          role="button"
          tabindex="0"
        >
          VISUAL
        </div>
        <div
          class="PillToggle--item"
          data-name="toggle-option-two"
          role="button"
          tabindex="0"
        >
          YAML
        </div>
      </div>
    </div>
  </div>
  <div
    class="Layout--vertical StyledProps--main"
  >
    <div
      class="OverlaySpinner--overlaySpinner"
    >
      <form
        class="FormikForm--main"
      >
        <div>
          <div
            class="StyledProps--font StyledProps--main main"
          >
            <div
              class="InputWithIdentifier--txtNameContainer TextInput--main"
            >
              <div
                class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
              >
                <div
                  class="Text--withDocsTooltip"
                  data-tooltip-id="deployEnv_identifier"
                >
                  <p
                    class="StyledProps--font StyledProps--main"
                  >
                    ID
                  </p>
                </div>
                <p
                  class="StyledProps--font StyledProps--main"
                >
                  :
                </p>
                <span
                  class="bp3-popover-wrapper"
                >
                  <span
                    class="bp3-popover-target"
                  >
                    <div
                      class="bp3-editable-text InputWithIdentifier--idValue"
                      tabindex="0"
                    >
                      <span
                        class="bp3-editable-text-content"
                        style="height: 1px; line-height: 1px; min-width: 0;"
                      >
                        qa
                      </span>
                    </div>
                  </span>
                </span>
                <span
                  class="bp3-icon StyledProps--main"
                  data-icon="Edit"
                  style="vertical-align: middle; cursor: pointer;"
                >
                  <svg
                    height="12"
                    viewBox="0 0 16 16"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                      fill="currentColor"
                      fill-rule="evenodd"
                      opacity="0.3"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="bp3-form-group"
                data-id="name-0"
              >
                <label
                  class="bp3-label"
                  for="name"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="deployEnv_name"
                  >
                    name
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="bp3-input-group"
                  >
                    <input
                      autocomplete="off"
                      class="bp3-input"
                      name="name"
                      type="text"
                      value="qa"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="StyledProps--font StyledProps--main"
              style="margin-bottom: 0px;"
            >
              <label
                class="Label--main bp3-label descriptionLabel"
              >
                optionalField
              </label>
              <div
                class="bp3-form-group"
                data-id="description-1"
              >
                <div
                  class="bp3-form-content"
                >
                  <textarea
                    autocomplete="off"
                    class="bp3-input bp3-fill"
                    maxlength="1024"
                    name="description"
                    placeholder="common.descriptionPlaceholder"
                  >
                    Decription mock
                  </textarea>
                </div>
              </div>
            </div>
            <div
              class="StyledProps--font StyledProps--main"
            >
              <label
                class="Label--main bp3-label descriptionLabel"
              >
                optionalField
              </label>
              <div
                class="bp3-form-group"
                data-id="undefined-2"
              >
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="bp3-input bp3-tag-input"
                  >
                    <div
                      class="bp3-tag-input-values"
                    >
                      <span
                        class="bp3-tag"
                        data-tag-index="0"
                      >
                        <span
                          class="bp3-text-overflow-ellipsis bp3-fill"
                        >
                          asd
                        </span>
                        <button
                          class="bp3-tag-remove"
                          type="button"
                        >
                          <span
                            class="bp3-icon bp3-icon-small-cross"
                            icon="small-cross"
                          >
                            <svg
                              data-icon="small-cross"
                              height="16"
                              viewBox="0 0 16 16"
                              width="16"
                            >
                              <desc>
                                small-cross
                              </desc>
                              <path
                                d="M9.41 8l2.29-2.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42L6.59 8 4.3 10.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l2.29 2.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                      <input
                        class="bp3-input-ghost"
                        data-mentions="kv-tag-input-tags"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
          >
            <label
              class="Label--main bp3-label label"
            >
              envType
            </label>
            <div
              class="bp3-form-group ThumbnailSelect--mainContainer thumbnailSelect"
              data-id="type-3"
            >
              <div
                class="bp3-form-content"
              >
                <div
                  class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                >
                  <label
                    class="Thumbnail--squareCardContainer Thumbnail--bigger"
                  >
                    <div
                      class="bp3-card bp3-interactive bp3-elevation-0 Card--card Thumbnail--squareCard Card--interactive"
                      tabindex="0"
                    >
                      <p
                        class="StyledProps--font StyledProps--main Thumbnail--label StyledProps--color StyledProps--color-black"
                      >
                        production
                      </p>
                    </div>
                    <input
                      name="type"
                      type="checkbox"
                      value="Production"
                    />
                  </label>
                  <label
                    class="Thumbnail--squareCardContainer Thumbnail--bigger"
                  >
                    <div
                      class="bp3-card bp3-interactive bp3-elevation-0 Card--card Thumbnail--squareCard Card--interactive"
                      tabindex="0"
                    >
                      <p
                        class="StyledProps--font StyledProps--main Thumbnail--label StyledProps--color StyledProps--color-black"
                      >
                        nonProduction
                      </p>
                    </div>
                    <input
                      name="type"
                      type="checkbox"
                      value="PreProduction"
                    />
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--padding-top-xlarge"
          >
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-primary"
              data-id="environment-save"
              type="submit"
            >
              <span
                class="bp3-button-text"
              >
                save
              </span>
            </button>
             
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-tertiary"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                cancel
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Environments should render Environments 2`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main yamlToggleEnv"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-flex-start"
      padding-top="8px"
    >
      <div
        class="PillToggle--optionBtns"
      >
        <div
          class="PillToggle--item"
          data-name="toggle-option-one"
          role="button"
          tabindex="0"
        >
          VISUAL
        </div>
        <div
          class="PillToggle--item PillToggle--selected"
          data-name="toggle-option-two"
          role="button"
          tabindex="0"
        >
          YAML
        </div>
      </div>
    </div>
  </div>
  <div
    class="Layout--vertical StyledProps--main"
  >
    <div
      class="OverlaySpinner--overlaySpinner"
    >
      <div
        class="StyledProps--font StyledProps--main"
      >
        <div />
        <div
          class="Layout--horizontal StyledProps--main StyledProps--padding-top-large"
        >
          <button
            class="bp3-button bp3-disabled bp3-loading Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-primary"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <div
              class="bp3-spinner bp3-button-spinner"
            >
              <div
                class="bp3-spinner-animation"
              >
                <svg
                  height="20"
                  stroke-width="16.00"
                  viewBox="-3.00 -3.00 106.00 106.00"
                  width="20"
                >
                  <path
                    class="bp3-spinner-track"
                    d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                  />
                  <path
                    class="bp3-spinner-head"
                    d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                    pathLength="280"
                    stroke-dasharray="280 280"
                    stroke-dashoffset="210"
                  />
                </svg>
              </div>
            </div>
            <span
              class="bp3-button-text"
            >
              save
            </span>
          </button>
             
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-tertiary"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              cancel
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
